src/ghci/stdin.rs

```diff
@@ -135,15 +137,64 @@ impl GhciStdin {
     }
 
     #[instrument(skip(self, stdout), level = "debug")]
-    pub async fn show_modules(&mut self, stdout: &mut GhciStdout) -> miette::Result<ModuleSet> {
+    pub async fn show_paths(&mut self, stdout: &mut GhciStdout) -> miette::Result<ShowPaths> {
         self.set_mode(stdout, Mode::Internal).await?;
 
         self.stdin
-            .write_all(b":show modules\n")
+            .write_all(b":show paths\n")
             .await
             .into_diagnostic()?;
 
-        stdout.show_modules().await
+        stdout.show_paths().await
+    }
+
+    #[instrument(skip(self, stdout), level = "debug")]
+    pub async fn show_targets(
+        &mut self,
+        stdout: &mut GhciStdout,
+        show_paths: &ShowPaths,
+    ) -> miette::Result<ModuleSet> {
+        self.set_mode(stdout, Mode::Internal).await?;
+
+        self.stdin
+            .write_all(b":show targets\n")
+            .await
+            .into_diagnostic()?;
+
+        stdout.show_targets(show_paths).await
+    }
+
+    #[instrument(skip(self, stdout), level = "debug")]
+    pub async fn eval(
+        &mut self,
+        stdout: &mut GhciStdout,
+        module: &str,
+        command: &GhciCommand,
+    ) -> miette::Result<()> {
+        self.set_mode(stdout, Mode::Internal).await?;
+
+        self.stdin
+            .write_all(format!(":module + *{module}\n").as_bytes())
+            .await
+            .into_diagnostic()?;
+        stdout.prompt(None).await?;
+
+        for line in command.as_ref().lines() {
+            self.stdin
+                .write_all(line.as_bytes())
+                .await
+                .into_diagnostic()?;
+            self.stdin.write_all(b"\n").await.into_diagnostic()?;
```

Comment:
minor suggestion:

```suggestion
            self.stdin
                .write_all(format!("{line}\n").as_bytes())
                .await
                .into_diagnostic()?;
```

Reply:
I was trying to avoid allocating another string by formatting it... it probably doesn't matter too much. Is doing multiple writes worse than allocating?